### ISSUE TITLE
Compilation failure when creating derived `Configs` for `HikariDataSource` bean

### ISSUE THREAD

**Issue by plm** [kxbmap/configs#21]
Using version Configs 0.4.4 and Scala 2.11.8, I'm trying to create a derived `Configs` bean for the `com.zaxxer.hikari.HikariDataSource` class as follows:
```
val hikariDataSourceConfigs = Configs.deriveBean[HikariDataSource]
```

This code produces the following error at compile time:
> Some[A] has no available apply methods

I tried the alternate form for deriving a bean where I provide the instance myself:
```
val hikariDataSourceConfigs = Configs.deriveBeanWith { new HikariDataSource }
```

SBT-based compilation of this code fails with a different error:
> A is not a class

IntelliJ-driven compilation reports this error as well, but additionally reports:
> Some cannot be instantiated because it does not conform to its self-type Some[A]

Running the SBT build with remote debugging enabled, I can see that in [`ConfigsMacro` on line 201](https://github.com/kxbmap/configs/blob/v0.4.4/core/src/main/scala/configs/macros/ConfigsMacro.scala#L201), the `ctx.target` is `A`, so this may explain the reported errors.

The `HikariDataSource` bean has various setters which take non-primitive types as their argument, such as [`setThreadFactory(ThreadFactory)`](https://github.com/brettwooldridge/HikariCP/blob/HikariCP-2.6.0/src/main/java/com/zaxxer/hikari/HikariConfig.java#L805). I don't have a `Configs[ThreadFactory]` in scope. If this is related to this failure, is it possible for the `BeanConfigsMacro` only generate code to invoke setters for which the requisite `Configs` type is in scope?

**Comment by kxbmap** [comment]
I think new `@ignoredBeanProperties` annotation can solve this problem.

In this case, you can write:
```scala
@ignoredBeanProperties("threadFactory", "logWriter", ...)
val hikariDataSourceReader = ConfigReader.derive[HikariDataSource]
```
(In 0.5.0, `Configs` renamed to `ConfigReader`)

**Comment by kxbmap** [comment]
Released 0.5.0, please check it out.
